### ISSUE TITLE
`doc_link_with_quotes` does not ignore code blocks

### ISSUE THREAD

**Issue by lopopolo** [rust-lang/rust-clippy#9321]
### Summary

Triple backtick code blocks are not ignored by this lint.

### Lint Name

doc_link_with_quotes

### Reproducer

I tried this code:

```rust
//! ENV is a hash-like accessor for environment variables.
//!
//! This module implements the [`ENV`] singleton object from Ruby Core.
//!
//! In Artichoke, the environment variable store is modeled as a hash map of
//! byte vector keys and values, e.g. `HashMap<Vec<u8>, Vec<u8>>`. Backends are
//! expected to convert their internals to this representation in their public
//! APIs. For this reason, all APIs exposed by ENV backends in this crate are
//! fallible.
//!
//! You can use this object in your application by accessing it directly. As a
//! Core API, it is globally available:
//!
//! ```ruby
//! ENV['PATH']
//! ENV['PS1'] = 'artichoke> '
//! ```
//!
//! [`ENV`]: https://ruby-doc.org/core-3.1.2/ENV.html

fn main() {
    println!("Hello, world!");
}
```

I saw this happen:

```
$ cargo clippy -- -W clippy::all -W clippy::pedantic
    Checking clippy_doc_link v0.1.0 (/Users/lopopolo/dev/artichoke/clippy_doc_link)
warning: possible intra-doc link using quotes instead of backticks
  --> src/main.rs:15:1
   |
15 | //! ENV['PATH']
   | ^^^^^^^^^^^^^^^
   |
   = note: `-W clippy::doc-link-with-quotes` implied by `-W clippy::pedantic`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#doc_link_with_quotes

warning: possible intra-doc link using quotes instead of backticks
  --> src/main.rs:16:1
   |
16 | //! ENV['PS1'] = 'artichoke> '
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#doc_link_with_quotes

warning: `clippy_doc_link` (bin "clippy_doc_link") generated 2 warnings
    Finished dev [unoptimized + debuginfo] target(s) in 0.43s
```

I expected to see this happen: These docs are inside a "triple backtick" code block with a `ruby` language tag. I expect this lint to ignore this text.

### Version

```text
rustc 1.63.0 (4b91a6ea7 2022-08-08)
binary: rustc
commit-hash: 4b91a6ea7258a947e59c6522cd5898e7c0a6a88f
commit-date: 2022-08-08
host: x86_64-apple-darwin
release: 1.63.0
LLVM version: 14.0.5
```

**Comment by jendrikw** [comment]
Dupe of #8961.
